data/model/repository.py

```diff
@@ -672,3 +676,60 @@ def mark_repository_for_deletion(namespace_name, repository_name, repository_gc_
         callback(namespace_name, repository_name)
 
     return marker.id
+
+
+def get_repository_size_and_cache(repo_id: int):
+    query = None
+    try:
+        query = (
+            RepositorySize.select(Repository.name, Repository.id, RepositorySize.size_bytes)
+            .join(Repository)
+            .where(Repository.id == repo_id)
+        ).get()
+    except RepositorySize.DoesNotExist:
+        force_cache_repo_size(repo_id)
+
+    output = query
+
+    return {
+        "repository_name": output.repository.name,
+        "repository_id": output.repository.id,
+        "repository_size": output.size_bytes,
+    }
+
+
+def get_size_during_upload(repo_id: int):
+    # TODO: Make this one trip to the db instead of 2?
+    size = get_repository_size_and_cache(repo_id)
+
+    query = (
+        BlobUpload.select(fn.Sum(BlobUpload.byte_count).alias("size_bytes")).where(
+            BlobUpload.repository_id == repo_id
+        )
+    ).get()
+
+    size_bytes = size["repository_size"] + query.size_bytes
+
+    return size_bytes
+
+
+def force_cache_repo_size(repo_id: int):
+    try:
+        cache = (
+            Manifest.select(fn.Sum(Manifest.layers_compressed_size).alias("size_bytes")).where(
+                Manifest.repository == repo_id
+            )
+        ).get()
+
+        size = cache.size_bytes
+    except Manifest.DoesNotExist:
+        size = get_size_during_upload(repo_id)
```

Comment:
This should either be 0 if the repo is empty, or the size of the already uploaded manifests. Wouldn't this be an issue for partially updated manifests (where blobs were uploaded but not the manifest)?